lib/safer_rails_console.rb

```diff
@@ -55,7 +58,36 @@ class Configuration
     end
 
     def set(**new_config)
+      CONFIG_DEFAULTS.each do |name, value|
+        env = get_value(value, name)
+        config[name] = env unless env.nil?
+      end
       config.merge!(new_config.select { |k, _v| CONFIG_DEFAULTS.key?(k) })
     end
+
+    private
+
+    def get_value(default, name)
+      env = ENV["#{PREFIX}#{name.upcase}"]
+
+      if env
+        if [true, false].include?(default)
+          return ActiveModel::Type::Boolean.new.cast(env)
+        elsif default.is_a?(::Array)
+          return env.split(' ')
+        elsif default.is_a?(::Hash)
+          return string_to_hash(env)
+        end
+
+        return env
+      end
+
+      nil
+    end
+
+    def string_to_hash(string)
```

Comment:
I wouldn't recommend handling "hash" arguments this way - gsubbing something that is structured can have very unpredictable side effects since the contents can be anything.